pkg/cmd/cli/cmd/set/volume.go

```diff
@@ -285,10 +289,20 @@ func (a *AddVolumeOptions) Validate(isAddOp bool) error {
 					return errors.New("must provide --path for --type=hostPath")
 				}
 			case "secret":
+				if len(a.DefaultMode) > 0 {
+					if ok, _ := regexp.MatchString(`\b0?[0-7]{3}\b`, a.DefaultMode); !ok {
```

Comment:
I think we can defer this to [api validation](https://github.com/kubernetes/kubernetes/blob/3a85ad951486f2f1047d1f57600aa52fa0edee88/pkg/api/validation/validation.go#L775-L778) which means you don't need to handle it here. You probably need to validate that the flag is not used with something that is not a secret or a configmap though.